### PR TITLE
Allow to override the Makefile's `DOCKER_MOUNT` variable

### PR DESCRIPTION
**Pull request by wk8**
**- What I did**

This patch provides some way for non-Linux devs to work on this repo without
suffering too much from the notorious terrible I/O perf of mounted volumes (eg
on OS X).

**- How I did it**

We now allow to override the Makefile's `DOCKER_MOUNT` variable through the env
variable of the same name.

The idea here is pretty simple: I/O perf on native mounted disks
on non-Linux (notably Mac OS) is just terrible, thus making it
a real pain to develop: one has to choose between re-building
the image after every single change (eg to run a test) or just
work directly inside the same container (eg with vim, but even then
one would have to re-configure their dev container every time
it gets destroyed - containers, after all, are not supposed to
be long-lived).

Allowing to override `DOCKER_MOUNT` makes it easy for everyone
to decide what their volume/syncing strategy is; for example
one can choose to use [docker-sync](https://github.com/EugenMayer/docker-sync)

This patch won't change anything for anyone who doesn't set the `DOCKER_MOUNT`
env variable in their environment; really tried to make it as little disrputive
as possible.

**- How to verify it**

The outputs of:
```
make -n shell
BIND_DIR=. make -n shell
```
should be unchanged compared to what it was before this patch; but:
```
DOCKER_MOUNT='-v test:/test' make -n shell
```
should replace all the `-v` flags from the commands above with just
`-v test:/test`, and finally:
```
DOCKER_MOUNT=' ' make -n shell
```
should be devoid of any `-v` flags.

**- A picture of a cute animal (not mandatory but encouraged)**

bunny

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Needs rebase

**Comment by wk8** [comment]
@AkihiroSuda : thanks, rebased

**Review by vdemeester — APPROVED** [review]
SGTM :frog:

**Comment by wk8** [comment]
@vdemeester : thanks for the review :) Is the next step to get another sign-off before we can merge?

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by AkihiroSuda** [comment]
LGTM

@thaJeztah @cpuguy83 PTAL?

**Comment by wk8** [comment]
@cpuguy83 honestly, I tried the caching options you mention, still got really, really poor perf.

This patch is not perfect, but at least it allows people to do their own thing if they want to/know what they're doing.

**Comment by AkihiroSuda** [comment]
ping @cpuguy83 @tianon @thaJeztah

**Review by thaJeztah — APPROVED** [review]
LGTM

but perhaps would be worth discussing with @djs55 about the performance issues you're seeing

**Pull request merged**
